frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,224 @@
+import { hasCPUandMemory } from './manage/ManageNodeResourceSection';
+
+export type WarningNotification = {
+  warningStatus: boolean;
+  title: string;
+  message: string;
+  isDefaultProfile: boolean;
+};
+
+export type HardwareProfileWarningMessage = {
+  warning: boolean;
+  title: string;
+  message: string;
+  isEnabled: boolean;
+};
 
 export const isHardwareProfileOOTB = (hardwareProfile: HardwareProfileKind): boolean =>
   hardwareProfile.metadata.labels?.['opendatahub.io/ootb'] === 'true';
+
+export const defaultNodes: Identifier[] = [
+  {
+    displayName: 'CPU',
+    identifier: 'cpu',
+    minCount: 1,
+    maxCount: 4,
+    defaultCount: 2,
+    resourceType: IdentifierResourceType.CPU,
+  },
+  {
+    displayName: 'Memory',
+    identifier: 'memory',
+    minCount: '2Gi',
+    maxCount: '8Gi',
+    defaultCount: '4Gi',
+    resourceType: IdentifierResourceType.MEMORY,
+  },
+];
+
+export const generateWarningTitle = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
+): string => {
+  if (!isEnabled) {
+    return 'All hardware profiles are disabled';
+  }
+  if (isAllWarnings) {
+    return 'All hardware profiles are invalid';
+  }
+  if (warningCount > 0) {
+    return 'One or more hardware profiles are invalid';
+  }
+  return '';
+};
+
+export const generateWarningMessage = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
+): string => {
+  if (!isEnabled) {
+    return 'You must have at least one hardware profile enabled for users to create workbenches or deploy models. Enable one or more profiles in the table below.';
+  }
+  if (isAllWarnings) {
+    return 'You must have at least one valid hardware profile enabled for users to create workbenches or deploy models. Take the appropriate actions below to re-validate your profiles.';
+  }
+  if (warningCount > 0) {
+    return 'One or more of your defined hardware profiles are invalid. Take the appropriate actions below to revalidate your profiles.';
+  }
+  return '';
+};
+
+export const generateWarningForHardwareProfiles = (
+  hardwareProfiles: HardwareProfileKind[],
+): HardwareProfileWarningMessage => {
+  let isEnabled = false;
+  let warningCount = 0;
+  for (const profile of hardwareProfiles) {
+    if (profile.spec.warning?.warningStatus === true) {
+      warningCount += 1;
+    }
+    if (profile.spec.enabled === true) {
+      isEnabled = true;
+    }
+  }
+
+  return {
+    warning: warningCount > 0 || !isEnabled,
+    title: generateWarningTitle(isEnabled, warningCount === hardwareProfiles.length, warningCount),
+    message: generateWarningMessage(
+      isEnabled,
+      warningCount === hardwareProfiles.length,
+      warningCount,
+    ),
+    isEnabled,
+  };
+};
+
+export const hardwareProfileWarning = (
+  hardwareProfiles: HardwareProfileKind[],
+): HardwareProfileKind[] =>
+  hardwareProfiles.map((profile) => {
+    let warningStatus = false;
+    const { identifiers } = profile.spec;
```

Comment:
There are so many `identifiers ?? []` below, you can actually do it here
```suggestion
    const identifiers = profile.spec.identifiers ?? [];
```